### DISCUSSION TITLE
Can't send local file for transcript in dart

### DISCUSSION THREAD

**Discussion by jgirardet** [deepgram/community#642] [General help] [answered]
Hi,
I tried hard to a send a file.mp3 or file.wav for transcript with dart.

I read some other disccussions and the blog post but it's always about websocket. I can't find any help anywhere on how to do the equivalent curl request.

Error: Bad Request: failed to process audio: corrupt or unsupported data", "request_id":"48e0851e-d81d-4355-a9dd-2881b0a82dcd"

```dart
  var data = FormData.fromMap({
    'files': [await MultipartFile.fromFile('sample.wav')],
  });

  final dio = Dio();
  final resp = await dio.post(
    "https://api.deepgram.com/v1/listen?model=nova-2&smart_format=true&language=fr",
    options: Options(
      headers: {
        "Authorization": "Token xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
      },
    ),
    data: data,
  );
```

the following works with curl:
```curl
curl   --request POST   --header 'Authorization: Token   xxxxx'  -T sample.wav  --url 'https://api.deepgram.com/v1/listen?model=nova-2&smart_format=true&language=fr'
```
curl works both with the mp3 or the wav version.

I tried to ue the http lib but same result

Does anyone has an idea ?

Thanks for help.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jgirardet** [comment, marked as answer]
Tsss, I swear I spent 2 days on it, I just made it after posting
.
Multipart.fromFile should not be used, just read the file :

```dart
data: await File("sample.mp3").readAsBytes(),
```
